File: Day-44/InsertionSort.py
def InsertionSort(arr):
    for i in range(1, len(arr)):
        temp = arr[i]
        j = i -1
        while(j>=0):
            if(arr[j]>temp):
                arr[j+1] = arr[j]
            else:
                break
            j-=1
        
        arr[j+1]=temp
    print(arr)

InsertionSort([10,1,7,6,14,9])

File: Day-44/InsertionSort.js
function InsertionSort(arr) {
  for (let i = 1; i < arr.length; i++) {
    let temp = arr[i];
    let j = i - 1;
    for (; j >= 0; j--) {
      if (arr[j] > temp) {
        //shift
        arr[j + 1] = arr[j];
      } else {
        break;
      }
    }
    arr[j + 1] = temp;
  }
  console.log(arr);
}

InsertionSort([10, 1, 7, 6, 14, 9]);
